packages/zoe/src/contracts/coveredCall.js

```diff
@@ -21,46 +25,48 @@ const rejectMsg = `The covered call option is expired.`;
 // { expirationDate, timerAuthority, underlyingAsset, strikePrice }
 
 // zcf is the Zoe Contract Facet, i.e. the contract-facing API of Zoe
-export const makeContract = harden(zcf => {
-  const { swap, assertKeywords, checkHook } = makeZoeHelpers(zcf);
-  assertKeywords(harden(['UnderlyingAsset', 'StrikePrice']));
+export const makeContract = harden(
+  /** @param {ContractFacet} zcf */ zcf => {
+    const { swap, assertKeywords, checkHook } = makeZoeHelpers(zcf);
+    assertKeywords(harden(['UnderlyingAsset', 'StrikePrice']));
 
-  const makeCallOptionInvite = sellerHandle => {
-    const {
-      proposal: { want, give, exit },
-    } = zcf.getOffer(sellerHandle);
+    const makeCallOptionInvite = sellerHandle => {
+      const {
+        proposal: { want, give, exit },
+      } = zcf.getOffer(sellerHandle);
 
-    const exerciseOptionHook = offerHandle =>
-      swap(sellerHandle, offerHandle, rejectMsg);
-    const exerciseOptionExpected = harden({
-      give: { StrikePrice: null },
-      want: { UnderlyingAsset: null },
-    });
+      const exerciseOptionHook = offerHandle =>
+        swap(sellerHandle, offerHandle, rejectMsg);
+      const exerciseOptionExpected = harden({
+        give: { StrikePrice: null },
+        want: { UnderlyingAsset: null },
+      });
 
-    return zcf.makeInvitation(
-      checkHook(exerciseOptionHook, exerciseOptionExpected),
-      'exerciseOption',
-      harden({
-        customProperties: {
-          expirationDate: exit.afterDeadline.deadline,
-          timerAuthority: exit.afterDeadline.timer,
-          underlyingAsset: give.UnderlyingAsset,
-          strikePrice: want.StrikePrice,
-        },
-      }),
-    );
-  };
+      return zcf.makeInvitation(
+        checkHook(exerciseOptionHook, exerciseOptionExpected),
+        'exerciseOption',
+        harden({
+          customProperties: {
+            expirationDate: exit.afterDeadline.deadline,
+            timerAuthority: exit.afterDeadline.timer,
```

Comment:
The one thing that's not better is that the access to afterDeadline (in VSCode) doesn't understand the disjunction and says  exit doesn't have an `afterDeadline` property.